training/src/app/store/reducers/app.reducer.ts

```diff
@@ -0,0 +1,29 @@
+import { AppState, initialAppState } from "../states/app.state";
+import { createReducer, on } from "@ngrx/store";
+import { AppActions } from "../actions/app.actions";
+
+const _appReducer = createReducer(
+	initialAppState,
+
+	on(AppActions.GET_RANDOM_JOKE_SUCCESS, (state, action): AppState => ({
+		...state,
+		randomJoke: action.payload
+	})),
+	on(AppActions.GET_RANDOM_JOKE_FAILED, (state, action): AppState => ({
+		...state,
+		randomJokeError: action.payload
+	})),
+
+	on(AppActions.GET_SEARCH_JOKES_SUCCESS, (state, action): AppState => ({
```

Comment:
if u decided to save searchJokesError when there is error, then on success u need to clean it

Reply:
same comment for random joke